api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -156,6 +158,19 @@ public static boolean selectsPartitions(
         caseSensitive);
   }
 
+  public static boolean isPartitionCol(BoundGroupBy expr, Table table) {
+    return table.specs().values().stream().allMatch(spec -> isPartitionCol(expr, spec));
+  }
+
+  public static boolean isPartitionCol(BoundGroupBy expr, PartitionSpec spec) {
```

Comment:
This isn't correct. This will return all partition fields for a given ID, but those fields may include a transform. For example, a field could represent `hours(ts)` where `ts` has id 3. In order to check whether this is a partition column you can group by directly, you'd need to check that the transform is the identity transform.